### PR TITLE
feat(components): [table] overflow tooltip supports `enterable` and customization

### LINKED ISSUE #4734: [Feature Request] Expect the mouse to move into the title area,when I use 'show-overflow-tooltip' in el-table-column.
### Existing Component
Yes

### Component Name
el-table

### Description
Expect the mouse to move into the title area,when I use 'show-overflow-tooltip' in el-table-column.

### LINKED ISSUE #7188: [Feature Request] [Enhancement] [table-column] show-overflow-tooltip 属性，希望增加复制内容功能
Feature type: **`Enhancement`**

## Details

- Enhanced compone nt name: `table-column`

## Description

**table-column** 的 **show-overflow-tooltip** 属性，能不能增加鼠标可以移上去**复制显示**的内容，而不是鼠标移开就消失？

### Additional comments

_(empty)_

### LINKED ISSUE #7277: [Feature Request] [Enhancement] [table-column] table组件的show-overflow-tooltip能否改进tooltip的消失方式
Feature type: **`Enhancement`**

## Details

- Enhanced compone nt name: `table-column`

## Description

开启show-overflow-tooltip后，单元格里的超长内容被截断隐藏了，完整的超长内容则会显示在单元格上方的tooltip里面，但是鼠标刚移上去就tooltip就消失了，请问我想把超长内容完整的复制下来应该怎么做？ 对于一个普通用户并不会开启开发者控制台定位这个元素。
是否可以改进tooltip的响应区域，在鼠标移入tooltip范围内时也保持显示，而不是移开表格的单元格就消失了呢？

### Additional comments

_(empty)_

### LINKED ISSUE #7565: [Feature Request] [Enhancement] [table-column] table的show-overflow-tooltip提供延缓关闭
Feature type: **`Enhancement`**

## Details

- Enhanced compone nt name: `table-column`

## Description

目前的table-column提供了show-overflow-tooltip使得内容过长时会冒泡显示，但是当鼠标一移开时冒泡的内容就消失了，无法对里面的内容进行复制。
希望提供delay、hide-after之类的冒泡延期消失功能，让我能复制它的全内容

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by plainheart**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

**Resolves issues**

- #7565
- #7277
- #7188
- #4734

**Related PRs**

- #9923

**Changes**

1. Add `tooltip-options` prop for `ElTable` to allow global configuration

2. The `show-overflow-tooltip` prop of `ElTableColumn` now accepts an object (the same as `tooltip-options` above) as the value

3. The overflow tooltip supports `enterable` and enabled by default

4. Removed the outdated prop `show-tooltip-when-overflow`

👉 [**▶ Playground**](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGVcbiAgICByZWY9XCJ0YWJsZVJlZlwiXG4gICAgOmRhdGE9XCJ0YWJsZURhdGFcIlxuICAgIHN0eWxlPVwid2lkdGg6IDEwMCVcIlxuICAgIHRvb2x0aXAtZWZmZWN0PVwibGlnaHRcIlxuICAgIDp0b29sdGlwLW9wdGlvbnM9XCJ7XG4gICAgICBwbGFjZW1lbnQ6ICdib3R0b20nLFxuICAgICAgb2Zmc2V0OiA4LFxuICAgICAgcG9wcGVyT3B0aW9uczoge1xuICAgICAgICBtb2RpZmllcnM6IFtcbiAgICAgICAgICB7XG4gICAgICAgICAgICBuYW1lOiAnY29tcHV0ZVN0eWxlcycsXG4gICAgICAgICAgICBvcHRpb25zOiB7XG4gICAgICAgICAgICAgIGdwdUFjY2VsZXJhdGlvbjogZmFsc2UsXG4gICAgICAgICAgICB9LFxuICAgICAgICAgIH0sXG4gICAgICAgIF1cbiAgICAgIH1cbiAgICB9XCJcbiAgPlxuICAgIDxlbC10YWJsZS1jb2x1bW5cbiAgICAgICBwcm9wPVwiZGF0ZVwiXG4gICAgICAgbGFiZWw9XCJEYXRlXCJcbiAgICAgICB3aWR0aD1cIjgwXCJcbiAgICAgICA6c2hvdy1vdmVyZmxvdy10b29sdGlwPVwieyBwbGFjZW1lbnQ6ICdyaWdodCcgfVwiXG4gICAgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uXG4gICAgICAgcHJvcD1cIm5hbWVcIlxuICAgICAgIGxhYmVsPVwiRGF0ZVwiXG4gICAgICAgd2lkdGg9XCIxMDBcIlxuICAgIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtblxuICAgICAgICBwcm9wPVwiYWRkcmVzc1wiXG4gICAgICAgIGxhYmVsPVwiQWRkcmVzczFcIlxuICAgICAgICB3aWR0aD1cIjE1MFwiXG4gICAgICAgIDpzaG93LW92ZXJmbG93LXRvb2x0aXA9XCJ7IGVudGVyYWJsZTogZmFsc2UsIHBvcHBlckNsYXNzOiAndGFibGUtdG9vbHRpcC1hZGRyZXNzJyB9XCJcbiAgICAvPlxuXHRcdDxlbC10YWJsZS1jb2x1bW5cbiAgICAgICBwcm9wPVwiYWRkcmVzc1wiXG4gICAgICAgbGFiZWw9XCJBZGRyZXNzMlwiXG4gICAgICAgd2lkdGg9XCIxNTBcIlxuICAgICAgIDpzaG93LW92ZXJmbG93LXRvb2x0aXA9XCJ7IGVmZmVjdDogJ2RhcmsnLCBwbGFjZW1lbnQ6ICd0b3AnLCBvZmZzZXQ6IDAsIHBvcHBlckNsYXNzOiAndGFibGUtdG9vbHRpcC1hZGRyZXNzMicgfVwiXG4gICAgLz5cbiAgPC9lbC10YWJsZT5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBFbFRhYmxlIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5pbnRlcmZhY2UgVXNlciB7XG4gIGRhdGU6IHN0cmluZ1xuICBuYW1lOiBzdHJpbmdcbiAgYWRkcmVzczogc3RyaW5nXG59XG5cbmNvbnN0IHRhYmxlUmVmID0gcmVmKClcblxuY29uc3QgdGFibGVEYXRhOiBVc2VyW10gPSBbXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMycsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMScsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wOCcsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNicsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wNycsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgYWRkcmVzczogJ05vLiAxODksIEdyb3ZlIFN0LCBMb3MgQW5nZWxlcycsXG4gIH0sXG5dXG48L3NjcmlwdD5cblxuPHN0eWxlPlxuLnRhYmxlLXRvb2x0aXAtYWRkcmVzcyB7XG4gIGZvbnQtc3R5bGU6IGl0YWxpYztcbiAgZm9udC13ZWlnaHQ6IGJvbGQ7XG4gIGJveC1zaGFkb3c6IHZhcigtLWVsLWJveC1zaGFkb3cpO1xufVxuLnRhYmxlLXRvb2x0aXAtYWRkcmVzczIge1xuICBib3gtc2hhZG93OiB2YXIoLS1lbC1ib3gtc2hhZG93KTtcbn1cbi50YWJsZS10b29sdGlwLWFkZHJlc3MyOmhvdmVyIHtcbiAgY29sb3I6IHZhcigtLWVsLWNvbG9yLXByaW1hcnkpIWltcG9ydGFudDtcbiAgY3Vyc29yOiBwb2ludGVyO1xuICB0ZXh0LWRlY29yYXRpb246IHVuZGVybGluZTtcbn1cbjwvc3R5bGU+IiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTEwOTg0LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTA5ODQtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9ICdodHRwczovL3ByZXZpZXctMTA5ODQtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctMTA5ODQtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MifX0=)

### CODE REVIEW

File: packages/components/table/src/util.ts [hunk 1]
```diff
@@ -345,37 +368,57 @@ export function createTablePopper(
...
       },
-      {
-        name: 'arrow',
-        options: {
-          element: arrow,
-          padding: 10,
-        },
+    })
+  }
+  if (arrow) {
```
Comment (chenxch) [inline packages/components/table/src/util.ts:402]: Can be merged with line:380

**Review by chenxch — COMMENTED** [review]
awesome~👍With the support of tooltipOptions, our form is more powerful.

It would be nice if this was reflected in the documentation, perhaps adding some test cases.

File: packages/components/table/src/util.ts [hunk 2]
```diff
@@ -347,7 +347,9 @@ export function createTablePopper(
   function renderContent(): HTMLDivElement {
     const isLight = tooltipOptions.effect === 'light'
     const content = document.createElement('div')
-    content.className = `${ns}-popper ${isLight ? 'is-light' : 'is-dark'}`
+    content.className = `${ns}-popper ${isLight ? 'is-light' : 'is-dark'} ${
+      tooltipOptions.popperClass || ''
+    }`
```
Comment (chenxch) [inline packages/components/table/src/util.ts:352]:
```
[`${ns}-popper`, `${isLight ? 'is-light' : 'is-dark'}`, `${tooltipOptions.popperClass || ''}`].join(' ')
```
Unnecessary spaces can be avoided.

Comment (plainheart) [reply packages/components/table/src/util.ts:352]: I just think the extra spaces won't bring side effects, so it's unnecessary to do something to remove it.

Comment (chenxch) [reply packages/components/table/src/util.ts:352]: I got it.🙈

**Review by chenxch — APPROVED** [review]

**Pull request merged**